tests/contrib/test_appengine.py

```diff
@@ -339,6 +417,27 @@ def setUp(self):
     def tearDown(self):
         self.testbed.deactivate()
 
+    def test_bad_ctor(self):
+        with self.assertRaises(ValueError):
+            StorageByKeyName(CredentialsModel, None, None)
```

Comment:
I'm a `py.test` believer, and in `py.test` the context manager is the one true way of doing this, so there's no ambiguity. (The other way is only there for 2.4 support).